### PR TITLE
Fix dynamic cluster integration tests

### PR DESCRIPTION
**Pull request by nsinkov**
## Changes proposed in this PR

- fix dynamic cluster integration tests

### CODE REVIEW

File: integration/tests/cook/test_dynamic_clusters.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                cluster["state-locked?"] = True
+                self.logger.info(f'Trying to update cluster {cluster}')
+                data, resp = util.update_compute_cluster(self.cook_url, cluster)
+                self.assertEqual(201, resp.status_code, resp.content)
+            # Then state = deleted
+            for cluster in clusters:
+                cluster["state"] = "deleted"
+                util.wait_until(lambda: util.update_compute_cluster(self.cook_url, cluster),
+                                lambda x: True if 201 == x[1].status_code else x[1].content,
+                                300000, 5000)
```
Comment (dposada) [inline integration/tests/cook/test_dynamic_clusters.py:53]: Why do we need the `wait_until` here?

File: integration/tests/cook/test_dynamic_clusters.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                "template": clusters[0]["template"]
+            }
+            data, resp = util.create_compute_cluster(self.cook_url, test_cluster)
+            self.assertEqual(201, resp.status_code, resp.content)
+
+        # Check that a job schedules successfully
+        command = "true"
+        job_uuid, resp = util.submit_job(self.cook_url, command=command, container=container)
+        self.assertEqual(201, resp.status_code, resp.content)
+        util.wait_for_instance(self.cook_url, job_uuid, status='success')
```
Comment (dposada) [inline integration/tests/cook/test_dynamic_clusters.py:70]: Can we also assert that the instance ran on the only running compute cluster?

File: integration/tests/cook/test_dynamic_clusters.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                            lambda x: True if 201 == x[1].status_code else x[1].content,
+                            300000, 5000)
+            # Hard-delete the original non-test clusters
+            for cluster in clusters:
+                self.logger.info(f'Trying to delete cluster {cluster}')
+                resp = util.delete_compute_cluster(self.cook_url, cluster)
+                self.assertEqual(204, resp.status_code, resp.content)
+            # Force give up leadership
+            self.logger.info(f"COOK_CHANGE_LEADER_COMMAND {os.getenv('COOK_CHANGE_LEADER_COMMAND')}")
+            cp = subprocess.run(os.getenv('COOK_CHANGE_LEADER_COMMAND'), shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (dposada) [inline integration/tests/cook/test_dynamic_clusters.py:90]: Why can't we just make the HTTP request to `/shutdown-leader` here?

File: integration/tests/cook/test_dynamic_clusters.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        # Old clusters should be re-created
+        # wait for cook to come up
+        util.wait_until(lambda: [cluster for cluster in util.compute_clusters(self.cook_url)['db-configs'] if cluster["state"] == "running"],
+                        lambda x: len(x) == len(clusters),
+                        420000, 5000)
+        # Check that a job schedules successfully
+        command = "true"
+        job_uuid, resp = util.submit_job(self.cook_url, command=command, container=container)
+        self.assertEqual(201, resp.status_code, resp.content)
+        util.wait_for_instance(self.cook_url, job_uuid, status='success')
```
Comment (dposada) [inline integration/tests/cook/test_dynamic_clusters.py:102]: Should we submit jobs until each of the old clusters has successfully run a job?

File: integration/tests/cook/util.py [hunk 5]
```diff
@@ -1935,12 +1939,22 @@ def kubernetes_settings():
     return settings(cook_url)['kubernetes']
 
 
-def create_compute_cluster(cook_url, cluster):
-    resp = session.post(f'{cook_url}/compute-clusters', json=cluster)
-    logger.info(f'create_compute_cluster resp: {resp.content}')
+def create_compute_cluster(cook_url, cluster, headers=None):
+    leader_url = scheduler_info_uncached(cook_url)['leader-url']
+    resp = session.post(f'{leader_url}/compute-clusters', json=cluster, headers=headers)
```
Comment (dposada) [inline integration/tests/cook/util.py:1944]: Should we just fix the redirect-to-leader so we don't have to do this?

File: integration/tests/cook/test_dynamic_clusters.py (hunk 4 above)
Comment (dposada) [reply integration/tests/cook/test_dynamic_clusters.py:110]: Why is it fine if one of them isn't working? What's the downside to checking that both are working?

File: integration/tests/cook/util.py (hunk 5 above)
Comment (dposada) [reply integration/tests/cook/util.py:1944]: OK, follow-up PR sounds good

File: integration/tests/cook/test_dynamic_clusters.py (hunk 4 above)
Comment (nsinkov) [reply integration/tests/cook/test_dynamic_clusters.py:110]: there's no reason to think that one will work but another won't
But if we want to test them all, how we get a job to schedule on each one?

Comment (dposada) [reply integration/tests/cook/test_dynamic_clusters.py:116]: I think you'd have to submit jobs until one of them gets matched to each compute cluster. Let me know what you think. If you think it's not worth the complexity, I'm fine with it.

Comment (nsinkov) [reply integration/tests/cook/test_dynamic_clusters.py:116]: we won't test here, we'll do it as a follow up test

**Review by dposada — APPROVED** [review]

**Pull request merged**